### PR TITLE
update ApplySecret to handle StringData

### PR DESCRIPTION
**Pull request by deads2k**
just make it work everywhere instead of just a single flow like https://github.com/openshift/library-go/pull/795

### CODE REVIEW

File: pkg/operator/resource/resourceapply/core.go [hunk 1]
```diff
@@ -234,10 +234,18 @@ func ApplyConfigMap(client coreclientv1.ConfigMapsGetter, recorder events.Record
...
-func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, required *corev1.Secret) (*corev1.Secret, bool, error) {
-	if len(required.StringData) > 0 {
-		return nil, false, fmt.Errorf("Secret.stringData is not supported")
+func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, requiredInput *corev1.Secret) (*corev1.Secret, bool, error) {
+	// copy the stringData to data.  Error on a data content conflict inside required.  This is usually a bug.
+	required := requiredInput.DeepCopy()
+	for k, v := range required.StringData {
+		if dataV, ok := required.Data[k]; ok {
+			if string(dataV) != v {
+				return nil, false, fmt.Errorf("Secret.stringData[%q] conflicts with Secret.data[%q]", k, k)
```
Comment (tnozicka) [inline pkg/operator/resource/resourceapply/core.go:243]: the api docs say this always overrides data. if you were to change a field in the code using stringData in the code (I don't see why), wouldn't this break the update?

Comment (deads2k) [reply pkg/operator/resource/resourceapply/core.go:243]: > the api docs say this always overrides data. if you were to change a field in the code using stringData in the code (I don't see why), wouldn't this break the update?

I think that would mean inconsistent information was provided as requiredInput, so a failure here is reasonable.

File: pkg/operator/resource/resourceapply/core.go [hunk 2]
```diff
@@ -234,10 +234,18 @@ func ApplyConfigMap(client coreclientv1.ConfigMapsGetter, recorder events.Record
...
+			}
+		}
+		required.Data[k] = []byte(v)
```
Comment (vareti) [inline pkg/operator/resource/resourceapply/core.go:246]: why is this not base64 encoded before writing into `Data`?

File: pkg/operator/resource/resourceapply/core.go [hunk 3]
```diff
@@ -234,10 +234,18 @@ func ApplyConfigMap(client coreclientv1.ConfigMapsGetter, recorder events.Record
 }
 
 // ApplySecret merges objectmeta, requires data
-func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, required *corev1.Secret) (*corev1.Secret, bool, error) {
-	if len(required.StringData) > 0 {
-		return nil, false, fmt.Errorf("Secret.stringData is not supported")
+func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, requiredInput *corev1.Secret) (*corev1.Secret, bool, error) {
+	// copy the stringData to data.  Error on a data content conflict inside required.  This is usually a bug.
+	required := requiredInput.DeepCopy()
```
Comment (vareti) [inline pkg/operator/resource/resourceapply/core.go:239]: just curios why requiredInput was not checked for `nil` value here? It seems to be the case previously too.

File: pkg/operator/resource/resourceapply/core.go (hunk 2 above)
Comment (deads2k) [reply pkg/operator/resource/resourceapply/core.go:246]: > why is this not base64 encoded before writing into `Data`?

encoding is done automatically during marshalling

File: pkg/operator/resource/resourceapply/core.go (hunk 3 above)
Comment (deads2k) [reply pkg/operator/resource/resourceapply/core.go:239]: > just curios why `requiredInput` was not checked for `nil` value here? It is the same case before you made any changes too.

uh, hopefully it panics at some point.  don't do that.

File: pkg/operator/resource/resourceapply/core.go (hunk 2 above)
Comment (vareti) [reply pkg/operator/resource/resourceapply/core.go:246]: Good to know. Thanks!

**Pull request merged**